### PR TITLE
Initial Services

### PR DESCRIPTION
**Pull request by sorja**
## Description
Added initial services

- #### Added/Removed/Updated something...

- SurveyService.get
- RecordService.get
- UserService.get
- UserResetPasswordRepository.hasValidResetPassword
- AuthGroupRepository.fetchUserGroups ==> AuthGroupRepository.getMany({userUuid})

- Added also .join().on()

## Types of changes

- [ ] Docs change
- [ ] Refactoring
- [ ] Dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Do you consider this PR needs further testing?

- [x] No
- [ ] Yes

## Disclaimer

- [ ] Todo: Add dbTransformCallback to some queries

### CODE REVIEW

**Comment by minotogna** [comment]
Task linked: [CU-hf7c13 Initial services](https://app.clickup.com/t/hf7c13)

File: src/db/dbs/transformCallback.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  // Assoc published and draft properties based on props
+  const currentRow = assocPublishedDraft ? _assocPublishedDraft(row) : row
+  const rowUpdated = Objects.camelize({ object: currentRow, skip: ['validation', 'props', 'props_draft'] })
+
+  if (!backup) {
+    return mergeProps({ row: rowUpdated, draft })
+  }
+
+  // backup: camelize props_draft column into propsDraft
+  rowUpdated.propsDraft = row.props_draft
```
Comment (minotogna) [inline src/db/dbs/transformCallback.ts:39]: what if the returned object doesn't need to include the propsDraft property?

File: src/db/sql/sqlSelectBuilder.ts [hunk 2]
```diff
@@ -51,6 +63,9 @@ export class SqlSelectBuilder extends SqlBuilder {
   build(): string {
     const parts: Array<string> = [`SELECT ${this._select.join(', ')}`, `FROM ${this._from.join(' ')}`]
 
+    if (!Objects.isEmpty(this._join) && !Objects.isEmpty(this._on)) {
+      parts.push(`JOIN ${this._join.join(' ')} ON ${this._on.join(' ')}`)
+    }
```
Comment (minotogna) [inline src/db/sql/sqlSelectBuilder.ts:68]: if we have several joins, won't this produce something like (where o is the source table):
```sql
join schema.tableA as a schema.tableB as b ON
o.a = a.a o.b = b.b 
```
would't we need to have a JoinConditionBuilder?

**Review by SteRiccio — APPROVED** [review]

File: src/db/sql/sqlJoinBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    this._on.push(...conditions)
+    return this
+  }
+
+  build(): string {
+    if (Objects.isEmpty(this._join) || Objects.isEmpty(this._on))
+      throw new Error(`missingParams: ${{ _join: this._join }}, ${{ _on: this._on }}`)
+    return `JOIN ${this._join.join(' ')} ON ${this._on.join(' ')}`
+  }
+}
```
Comment (minotogna) [inline src/db/sql/sqlJoinBuilder.ts:24]: this is single join. SqlSelectBuilder has an array of sqlJoinBuilders

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by minotogna — DISMISSED** [review]

**Review by minotogna — APPROVED** [review]

**Review by SteRiccio — APPROVED** [review]

**Pull request merged**